core/src/main/scala/org/apache/spark/scheduler/ResultTask.scala

```diff
@@ -65,4 +65,6 @@ private[spark] class ResultTask[T, U](
   override def preferredLocations: Seq[TaskLocation] = preferredLocs
 
   override def toString = "ResultTask(" + stageId + ", " + partitionId + ")"
+
+  override def canEqual(other: Any): Boolean = other.isInstanceOf[ResultTask[T, U]]
```

Comment:
So `equals` is not overridden in these subclasses because equality does not depend on their additional fields? just checking that this is definitely desirable.